### DISCUSSION TITLE
I dont understand how to use the interface options.

### DISCUSSION THREAD

**Discussion by J0914** [outline/rich-markdown-editor#509] [Q&A]
First of all, this editor is amazing and it has made my project SO much better, that said, I cannot figure out how to use the getHeadings interface? I dont understand how to use it, I can't find a codepen with an example and it's the only thing left that I need from this editor. Can anyone show me an example of how to use it, and how to get information from it, hopefully in the next couple of days before my project is due? thank you!

**Comment by tommoor** [comment]
Pass a [ref prop](https://reactjs.org/docs/refs-and-the-dom.html) to the editor and then call `getHeadings()` on the ref, an example here:

https://github.com/outline/outline/blob/a81fbd8608ec3322649affad1002c30fcb581711/app/scenes/Document/components/Document.js#L322

**Reply by J0914** [reply]
Thank you so much for this, you saved the day!
